libcaf_core/caf/flow/op/mcast.hpp

```diff
@@ -192,6 +204,23 @@ class mcast : public hot<T> {
     }
   }
 
+  // -- implementation of ucast_sub_state_listener -----------------------------
+
+  void on_disposed(state_type* ptr) final {
+    super::ctx_->delay_fn([mc = strong_this(), sptr = state_ptr_type{ptr}] {
+      if (auto i = std::find(mc->states_.begin(), mc->states_.end(), sptr);
+          i != mc->states_.end()) {
+        // We don't care about preserving the order of elements in the vector.
+        // Hence, we can swap the element to the back and then pop it.
+        auto last = mc->states_.end() - 1;
+        if (i != last)
+          std::swap(*i, *last);
```

Comment:
This could be just `*i = std::move(*last)` since an `intrusive_ptr` implements its move assignment as pointer swap.